### PR TITLE
fix 'Very slow proxing for http://www.answers.com' (close#657)

### LINKED ISSUE #657: Very slow proxing for http://www.answers.com
see https://testcafe-hhhm.devexpress.com/history?timestamp=1467659469614&siteUrl=http://www.answers.com/.
It happens because we cyclically send `set-cookie` service message. (see video - http://screencast.com/t/vBYd5NfW3j).
Need to investigate why we do this.

### PR DESCRIPTION
**Pull request by miherlosev**
see discussion in #657  for more details.
@churkin @inikulin 

Pull request changes:
1)fix #657 
2)remove unnecessary `processDomMeth` calls from tests

### CODE REVIEW

File: src/client/utils/throttle.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function throttle (func, ms) {
+    var isSuspend = false;
+    var savedArgs = null;
+    var savedThis = null;
+
+    function wrapper () {
+        if (isSuspend) {
+            savedArgs = arguments;
+            savedThis = this;
```
Comment (inikulin) [inline src/client/utils/throttle.js]: I'd rather go without saving context, expecting that `func` is already bound

File: src/client/utils/throttle.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            return;
+        }
+
+        func.apply(this, arguments);
+
+        isSuspend = true;
```
Comment (inikulin) [inline src/client/utils/throttle.js]: `suspended`

**Comment by inikulin** [comment]
\r-

File: src/client/sandbox/cookie.js [hunk 3]
```diff
@@ -129,12 +142,11 @@ export default class CookieSandbox extends SandboxBase {
...
         var setCookieMsg = {
             cmd:    COMMAND.setCookie,
-            cookie: value,
+            cookie: this.getCookie(),
             url:    document.location.href
         };
 
-        // NOTE: Meanwhile, synchronize cookies with the server cookie jar.
-        transport.queuedAsyncServiceMsg(setCookieMsg);
+        this.throttledCookieSync(setCookieMsg);
```
Comment (inikulin) [inline src/client/sandbox/cookie.js]: it should be named `syncCookies`, `Sync` suffix means that functions has `sync` nature, but we need to synchronize cookies here, also we can go without `throttled`, it's just an implementation details.

File: src/client/utils/throttle.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import nativeMethods from '../sandbox/native-methods';
+
+export default function throttle (func, ms) {
+    var isSuspend = false;
```
Comment (inikulin) [inline src/client/utils/throttle.js]: I believe it's a little bit more complicated then that. We set cookie for particular URL, what if we throttle setting cookies for particular URL and page unload happens? I guess we should perform additional synchronization on before unload.

Comment (inikulin) [inline src/client/utils/throttle.js]: We need tests for these scenarios as well

**Comment by inikulin** [comment]
\r-

File: src/client/utils/throttle.js (hunk 4 above)
Comment (miherlosev) [inline src/client/utils/throttle.js]: After an additional investigation I found that cookie synchronization every <value> ms is a bad choise.
We can get out of cookie sync for the following case:
```javascript
document.cookie = 'key1=value1';
document.cookie = 'key2=value2';
xhrRequest with cookie (or another request with cookie).
```

**Comment by inikulin** [comment]
We can attach cookies with xhr in that case in special hader

File: src/client/utils/throttle.js (hunk 4 above)
Comment (inikulin) [inline src/client/utils/throttle.js]: We can attach cookies with xhr in that case if synchronization is not done yet

Comment (miherlosev) [inline src/client/utils/throttle.js]: The task is more complicated.
We need an addition cookie sync for the following cases (if it is not done yet):
1)XmlHttpRequest and Fetch requests (store in custom header)
2)Images, script and style loading (pass via additional url parameter)
3)Page unload (store into custom input)

Waiting for https://github.com/DevExpress/testcafe-hammerhead/pull/721

Comment (miherlosev) [inline src/client/utils/throttle.js]: I think we can completely remove throttling and `setCookie` service messages and
perform cookie synchronization with server only for cases from list above.

Comment (inikulin) [inline src/client/utils/throttle.js]: Not sure about it

**Comment by miherlosev** [comment]
We will rethink cookie processing mechanism -https://github.com/DevExpress/testcafe-hammerhead/issues/1137

**Pull request closed without merging**